types/aws-lambda/trigger/cloudwatch-alarm.d.ts

```diff
@@ -0,0 +1,69 @@
+import { Handler } from "../handler";
+
+export interface CloudWatchAlarmState {
+    value: string;
+    reason: string;
+    timestamp: string;
+    reasonData?: string;
+    actionsSuppressedBy?: string;
+    actionsSuppressedReason?: string;
+}
+
+export interface CloudWatchMetric {
+    namespace: string;
+    name: string;
+    dimensions?: Record<string, string>;
+}
+
+export interface CloudWatchMetricStat {
+    metric: CloudWatchMetric;
+    period: number;
+    stat: string;
+    unit?: string;
+}
+
+export interface CloudWatchAlarmMetric {
+    id: string;
+    metricStat: CloudWatchMetricStat;
+    returnData: boolean;
+}
+
+export interface CloudWatchAlarmExpression {
+    id: string;
+    expression: string;
+    label: string;
+    returnData: boolean;
+}
+
+export type CloudWatchAlarmMetricDataQuery = CloudWatchAlarmMetric | CloudWatchAlarmExpression;
+
+export interface CloudWatchAlarmConfiguration {
+    metrics: CloudWatchAlarmMetricDataQuery[];
+    description?: string;
+}
+
+export interface CloudWatchAlarmCompositeConfiguration {
+    alarmRule: string;
+    description?: string;
+    actionsSuppressor?: string;
+    actionsSuppressorWaitPeriod?: number;
+    actionsSuppressorExtensionPeriod?: number;
+}
+
+export interface CloudWatchAlarmData {
+    alarmName: string;
+    state: CloudWatchAlarmState;
+    previousState: CloudWatchAlarmState;
+    configuration: CloudWatchAlarmConfiguration | CloudWatchAlarmCompositeConfiguration;
+}
+
+export interface CloudWatchAlarmEvent {
+    source: string;
+    alarmArn: string;
+    accountId: string;
+    time: string;
+    region: string;
+    alarmData: CloudWatchAlarmData;
+}
+
+export type CloudWatchAlarmHandler = Handler<CloudWatchAlarmEvent, void>;
```

Comment:
Nit-iest of nits, but we try to have the Handler type at the top, followed by the Event (and Result if present).

Reply:
No worries, happy to follow the convention. I've pushed an update.